daisy/step_wait_for_instances_signal.go

```diff
@@ -178,6 +192,61 @@ func waitForSerialOutput(s *Step, project, zone, name string, so *SerialOutput,
 	}
 }
 
+func waitForGuestAttribute(s *Step, project, zone, name string, ga *GuestAttribute, interval time.Duration) DError {
```

Comment:
This is very similar to `waitForSerialOutput` except for reading/comparing values. Consider extracting the common logic in one func and pass specific handlers as e.g. params.

Reply:
they are only superficially similar

Reply:
In terms of refactoring, a good shared candidate is polling for instance status. The version here looks at the 404 code, for example, but the serial output doesn't. So it'd be nice to have a single version that followed the same logic.

Reply:
i am not following either of you. the only options to factor out common code here as i tried it are much, much worse. to me it seems this is the minimum unit, since the logic is actually fundamentally different between them.

Reply:
I'll defer to Zoran on this